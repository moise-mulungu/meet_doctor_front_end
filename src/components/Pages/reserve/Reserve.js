import React from 'react';
import './style.css';
import { useForm } from 'react-hook-form';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { createReservation } from '../../../redux/reservations/reserve';

const ReservationForm = () => {
  const dispatch = useDispatch();
  const { data } = useSelector((state) => state.MostRecent);
  const navigate = useNavigate();
  /* eslint-disable */
   const { register, handleSubmit } = useForm();
   /* eslint-disable */
  const onSubmit = (data) => (dispatch(createReservation(data)) ? navigate('/main') : null);
  return (

    
    <div className='body'>

      <div className='R-Form'>
          
          <div className='form  form-group'>
             <h3>Book  your appointment</h3>
              
                  <form onSubmit={handleSubmit(onSubmit)}>
                    
                        {  /* register your input into the hook by invoking the "register" function */}
                         
                            <lable className="frst-L">Date
                            <lable >City</lable> 
                            <lable >Select a doctor</lable> 
                            </lable>
                            
                                    <input {...register('datetime')} type="date"   />              
                                <input {...register('city')} type="text" placeholder="City" className='city'/>
                                
                                <select {...register('doctor_id')} >
                          
                                  {data.map((doctor) => (
                                  <option key={doctor.id} value={doctor.id}>
                                  {doctor.name}
                                  </option>
                                ))}
                            </select>
                  
                
                    <button className='btn '>Book Now</button>
                    
                 </form>
             </div>
          
       </div>
   </div>
  );
};
export default ReservationForm;