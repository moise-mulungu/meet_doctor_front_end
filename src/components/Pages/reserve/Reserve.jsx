import React from 'react';
import './style.css';
import { useForm } from 'react-hook-form';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { createReservation } from '../../../redux/reservations/reserve';

const ReservationForm = () => {
  const dispatch = useDispatch();
  const { data } = useSelector((state) => state.MostRecent);
  const navigate = useNavigate();
  /* eslint-disable */
   const { register, handleSubmit } = useForm();
   /* eslint-disable */
  const onSubmit = (data) => (dispatch(createReservation(data)) ? navigate('/main') : null);
  return (

    
    <div className='body'>

      <div className='R-Form'>
          
          <div className='form  form-group'>
            <h3>Create your appointment</h3>
                
            <form onSubmit={handleSubmit(onSubmit)}>

              {  /* register your input into the hook by invoking the "register" function */}
                <label> Date </label>
                <div className='mb-2'>
                   <input {...register('datetime')} type="date" className='form-control'  />
                  </div>
                  <label> City</label>
                  <div  className='mb-2'>
                   <input {...register('city')} type="text" placeholder="City" className='form-control'/>
                  </div>
                  <label>  Select a Doctor</label>
                  <div  className='mb-2'>
                   <select {...register('doctor_id')} className='form-control'>
                
                   {data.map((doctor) => (
                    <option key={doctor.id} value={doctor.id}>
                        {doctor.name}
                        </option>
                      ))}
                   </select>
                   </div>
                  <div  className='mb-2'>
                   <button className='btn btn-primary btn-block'>Reserve</button>
                  </div>
            </form>
          </div>
       </div>
   </div>
  );
};
export default ReservationForm;
