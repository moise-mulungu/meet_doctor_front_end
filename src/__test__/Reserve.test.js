import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import Reserve from '../components/Pages/reserve/Reserve'


it('Check if Reserve page renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <Reserve />
      </BrowserRouter>
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});

afterEach(() => {
  jest.clearAllMocks();
});