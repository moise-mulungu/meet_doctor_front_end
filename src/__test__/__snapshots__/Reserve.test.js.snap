// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if Reserve page renders correctly 1`] = `
<div
  className="body"
>
  <div
    className="R-Form"
  >
    <div
      className="form  form-group"
    >
      <h3>
        Book  your appointment
      </h3>
      <form
        onSubmit={[Function]}
      >
        <lable
          className="frst-L"
        >
          Date
          <lable>
            City
          </lable>
          <lable>
            Select a doctor
          </lable>
        </lable>
        <input
          name="datetime"
          onBlur={[Function]}
          onChange={[Function]}
          type="date"
        />
        <input
          className="city"
          name="city"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="City"
          type="text"
        />
        <select
          name="doctor_id"
          onBlur={[Function]}
          onChange={[Function]}
        />
        <button
          className="btn "
        >
          Book Now
        </button>
      </form>
    </div>
  </div>
</div>
`;
